### PR TITLE
ProductsController v2 

### PR DESCRIPTION
**Pull request by AntonPuko**

### CODE REVIEW

File: src/repository/MongoDb.js [hunk 1]
```diff
@@ -45,11 +45,18 @@ class MongoDb extends BaseMongoDb implements IBaseDatabase {
     await this.__runForCollection(
       collectionName,
       async (collection: Object): Promise<*> => {
-        const { page, pageSize = 25, ...otherQuery } = query;
+        const { page, pageSize = 25, skip, take, ...otherQuery } = query;
```
Comment (jlkalberer) [inline src/repository/MongoDb.js]: Hmm.. we already have `page` and `pageSize`. Are these just for older APIs? You can probably give `take` a default value

File: src/repository/NeDb.js [hunk 2]
```diff
@@ -52,14 +52,21 @@ class NeDb extends BaseMongoDb implements IBaseDatabase {
     await this.__runForCollection(
       collectionName,
       async (collection: Object): Promise<*> => {
-        const { page, pageSize = 25, ...otherQuery } = query;
-        let boundFunction = collection.find(otherQuery);
+        const { page, pageSize = 25, skip, take, ...otherQuery } = query;
```
Comment (jlkalberer) [inline src/repository/NeDb.js]: Same here

File: src/repository/MongoDb.js [hunk 3]
```diff
@@ -45,7 +45,7 @@ class MongoDb extends BaseMongoDb implements IBaseDatabase {
     await this.__runForCollection(
       collectionName,
       async (collection: Object): Promise<*> => {
-        const { page, pageSize = 25, skip, take, ...otherQuery } = query;
+        const { skip = 25, take, ...otherQuery } = query;
```
Comment (jlkalberer) [inline src/repository/MongoDb.js]: nit: `take` should be 25. Skip should be null

Comment (AntonPuko) [reply src/repository/MongoDb.js]: ops

**Pull request merged**